ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/LogSegmentPath.java

```diff
@@ -58,9 +58,28 @@ public LogSegmentStartEnd getStartEnd() {
   }
 
   @Override
-  @SuppressFBWarnings("EQ_COMPARETO_USE_OBJECT_EQUALS")
   public int compareTo(LogSegmentPath that) {
-    return Comparator.comparing(LogSegmentPath::getStartEnd).compare(this, that);
+    return Comparator
+        .comparing(LogSegmentPath::getPath)
```

Comment:
We should not compare paths since open segment and closed segment have different path name formats which may not follow the log index order.